src/mocompiler.ts

```diff
@@ -1,42 +1,19 @@
 import encoding from 'encoding';
 import { HEADERS, formatCharset, generateHeader, compareMsgid } from './shared.js';
 import contentType from 'content-type';
+import {GetTextTranslation, GetTextTranslations, Size, TranslationBuffers, Translations} from "./types.js";
+import { Transform } from 'readable-stream';
 
-/**
- * @typedef {import('node:stream').Transform} Transform
- * @typedef {import('./types.js').GetTextTranslation} GetTextTranslation
- * @typedef {import('./types.js').GetTextTranslations} GetTextTranslations
- * @typedef {import('./types.js').Translations} Translations
- * @typedef {import('./types.js').WriteFunc} WriteFunc
- */
-
-/**
- * @typedef {Object} Size Data about the size of the compiled MO object.
- * @property {number} msgid The size of the msgid section.
- * @property {number} msgstr The size of the msgstr section.
- * @property {number} total The total size of the compiled MO object.
- */
-
-/**
- * @typedef {{ msgid: Buffer, msgstr: Buffer }} TranslationBuffers A translation object partially parsed.
- */
-
-/**
- *
- * @typedef {Object} CompilerOptions MO compiler options
- * @property {'be'|'le'} [endian='le'] Endianness of the output buffer. Default is 'le'
- */
 
 /**
  * Exposes general compiler function. Takes a translation
  * object as a parameter and returns binary MO object
  *
- * @param {GetTextTranslations} table Translation object
- * @param {CompilerOptions} [options] MO compiler options
- * @return {Buffer} Compiled binary MO object
+ * @param table Translation object
+ * @return Compiled binary MO object
  */
-export default function (table, options = { endian: 'le' }) {
-  const compiler = new Compiler(table, options);
+export default function (table: GetTextTranslations): Buffer {
+  const compiler = new Compiler(table);
```

Comment:
@johnhooks, it's time to update also this to classes, isn't it?

seems that TypeScript expects a new expression to be used with a class or a function with a constructor signature.

Reply:
I agree, let's do the transition to classes too.

Reply:
I tried that but I have to go back on my steps... converting to class now would also mean invalidating the diff and losing the ability to compare it with the current version. Temporarily I will use `@ts-ignore` to work around the problem, converting to class will be the next step